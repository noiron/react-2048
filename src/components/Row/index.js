import React from 'react';
import PropTypes from 'prop-types';
import Cell from '../Cell';
import styles from '../Board/board.css';

export default class Row extends React.Component {
    static propTypes = {
        row: PropTypes.arrayOf(PropTypes.number).isRequired
    };

    render() {
        const { row } = this.props;

        return (
            <div className={styles.boardRow}>
                { row.map((num, index) => <Cell value={num} key={index} />)}
            </div>
        )
    }
}
