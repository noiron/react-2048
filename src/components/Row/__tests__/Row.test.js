import React from 'react'
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Row from '..';

Enzyme.configure({ adapter: new Adapter() });

describe('<Row />', () => {
    it('component render', () => {
        const matrix = [[0, 0, 4, 4], [0, 2, 0, 0], [4, 0, 0, 0], [0, 0, 0, 0]];
        matrix.forEach(r => {
            const row = renderer.create(<Row row={r} />).toJSON();
            expect(row).toMatchSnapshot();
        })
    })
})
