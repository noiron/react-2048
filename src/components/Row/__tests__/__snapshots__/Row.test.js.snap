// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> component render 1`] = `
<div
  className={undefined}
>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    >
      4
    </div>
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    >
      4
    </div>
  </div>
</div>
`;

exports[`<Row /> component render 2`] = `
<div
  className={undefined}
>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    >
      2
    </div>
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
</div>
`;

exports[`<Row /> component render 3`] = `
<div
  className={undefined}
>
  <div
    className=" "
  >
    <div
      className={undefined}
    >
      4
    </div>
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
</div>
`;

exports[`<Row /> component render 4`] = `
<div
  className={undefined}
>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
  <div
    className=" "
  >
    <div
      className={undefined}
    />
  </div>
</div>
`;
